src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/SampleValidateHelper.java

```diff
@@ -8,15 +8,12 @@
 
 public class SampleValidateHelper {
 
-  public static void validateNewValue(
-      Entry<String, String> entry, ColumnValidator columnValidator, EventType eventType) {
-    if (columnValidator.getColumnName().equals(entry.getKey())) {
-      Map<String, String> validateThis = Map.of(entry.getKey(), entry.getValue());
+  public static void validateNewValue(String columnName, String validateThis, ColumnValidator columnValidator, EventType eventType) {
 
-      Optional<String> validationErrors = columnValidator.validateRow(validateThis);
+    if (columnValidator.getColumnName().equals(columnName)) {
+      Optional<String> validationErrors = columnValidator.validateData(validateThis, true);
       if (validationErrors.isPresent()) {
-        throw new RuntimeException(
-            eventType + " failed validation for column name: " + columnValidator.getColumnName());
+        throw new RuntimeException(eventType + " event: " + validationErrors.get());
```

Comment:
again, we are missing an opportunity here to collect all the validation problems and then throw an almighty error at the end